tests/testthat/test-type-sf.R

```diff
@@ -213,22 +214,31 @@ test_that("`precision` and `crs` attributes of `sfc` vectors are restored", {
 	expect_identical(st_crs(x), st_crs(out))
 })
 
-test_that("`precision` and `crs` attributes of `sfc` vectors are combined", {
+test_that("`precision` attributes of `sfc` vectors are combined", {
 	x = st_sfc(st_point(c(pi, pi)), precision = 1e-4, crs = 3857)
 	y = st_sfc(st_point(c(0, 0)), precision = 1e-4, crs = 3857)
 
 	out = vctrs::vec_c(x, y)
 	expect_identical(st_precision(x), st_precision(out))
-	expect_identical(st_crs(x), st_crs(out))
 
-        # These used to be errors before we fell back to c()
+	# These used to be errors before we fell back to c()
 	y = st_sfc(st_point(c(0, 0)), precision = 1e-2, crs = 3857)
 	expect_identical(vctrs::vec_c(x, y), c(x, y))
 	# expect_error(vctrs::vec_c(x, y), "precisions not equal")
+})
 
-	y = st_sfc(st_point(c(0, 0)), precision = 1e-4, crs = 4326)
-	expect_identical(vctrs::vec_c(x, y), c(x, y))
-	# expect_error(vctrs::vec_c(x, y), "coordinate reference systems not equal")
+test_that("`crs` attributes of `sfc` vectors must be the same", {
+  x = st_sfc(st_point(c(pi, pi)), precision = 1e-4, crs = 3857)
+  y = st_sfc(st_point(c(0, 0)), precision = 1e-4, crs = 3857)
+
+  out = vctrs::vec_c(x, y)
+  expect_identical(st_crs(x), st_crs(out))
+
+  # Error on different `crs` comes from sf as of 1.0-10
+  y = st_sfc(st_point(c(0, 0)), precision = 1e-4, crs = 4326)
+  expect_snapshot(error = TRUE, {
+    vctrs::vec_c(x, y)
+  })
```

Comment:
I can't seem to figure out how to get rstudio to respect the tab width thing for just one file